### PR TITLE
PR: Add an option to execute IPython events while debugging

### PR DESCRIPTION
**Pull request by impact27**
## Description of Changes

Add setting for `pdb_execute_events`

This allow enabling/disabling pre- and post-execute events. Notably inline plots are shown.

* [x] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [x] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

Depends on spyder-ide/spyder-kernels#175

### Issue(s) Resolved

Fixes #

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:
impact27

video

### CODE REVIEW

**Comment by ccordoba12** [comment]
This is pretty cool, thanks @impact27! But why do we need this as an option and not make it the default in spyder-kernels?

**Comment by impact27** [comment]
You need to disable it to debug code that creates figures for example.

**Comment by ccordoba12** [comment]
This has a conflict with master. Please solve it so I can merge it.

File: continuous_integration/azure/install.bat [hunk 1]
```diff
@@ -34,7 +34,7 @@ if %USE_CONDA% == yes (
 )
 
 :: Install spyder-kernels from master
-pip install -q --no-deps git+https://github.com/spyder-ide/spyder-kernels
+pip install -q --no-deps git+https://github.com/spyder-ide/spyder-kernels.git@refs/pull/175/head
```
Comment (ccordoba12) [inline continuous_integration/azure/install.bat]: Please revert these changes because I just merged spyder-ide/spyder-kernels#175.
